components/lock.rst

```diff
@@ -0,0 +1,257 @@
+============================================  ======  ========  ========
+
+.. _lock-store-flock:
+
+FlockStore
+~~~~~~~~~~
+
+The FlockStore uses the file system on the local computer to create the locks.
+It does not support expiration, but the lock is automatically released when the
+PHP process is terminated::
+
+    use Symfony\Component\Lock\Store\FlockStore;
+
+    // the argument is the path of the directory where the locks are created
+    $store = new FlockStore(sys_get_temp_dir());
+
+.. caution::
+
+    Beware that some file systems (such as some types of NFS) do not support
+    locking. In those cases, it's better to use a local file or a remote store
+    based on Redis or Memcached.
+
+.. _lock-store-memcached:
+
+MemcachedStore
+~~~~~~~~~~~~~~
+
+The MemcachedStore saves locks on a Memcached server, so first you must create
+a Memcached connection implements the ``\Memcached`` class. This store does not
+support blocking, and expects a TTL to avoid stalled locks::
+
+    use Symfony\Component\Lock\Store\MemcachedStore;
+
+    $memcached = new \Memcached();
+    $memcached->addServer('localhost', 11211);
+
+    $store = new MemcachedStore($memcached);
+
+.. note::
+
+    Memcached does not support TTL lower than 1 second.
+
+.. _lock-store-redis:
+
+RedisStore
+~~~~~~~~~~
+
+The RedisStore saves locks on a Redis server, so first you must create a Redis
+connection implements the ``\Redis``, ``\RedisArray``, ``\RedisCluster`` or
+``\Predis`` classes. This store does not support blocking, and expects a TTL to
+avoid stalled locks::
+
+    use Symfony\Component\Lock\Store\RedisStore;
+
+    $redis = new \Redis();
+    $redis->connect('localhost');
+
+    $store = new RedisStore($redis);
+
+.. _lock-store-semaphore:
+
+SemaphoreStore
+~~~~~~~~~~~~~~
+
+The SemaphoreStore uses the `PHP semaphore functions`_ to create the locks::
+
+    use Symfony\Component\Lock\Store\SemaphoreStore;
+
+    $store = new SemaphoreStore();
+
+.. _lock-store-combined:
+
+CombinedStore
+~~~~~~~~~~~~~
+
+The CombinedStore is designed for High Availability applications because it
+manages several stores in sync (for example, several Redis servers). When a lock
+is being acquired, it forwards the call to all the managed stores, and it
+collects their responses. If a simple majority of stores have acquired the lock,
+then the lock is considered as acquired; otherwise is not acquired::
+
+    use Symfony\Component\Lock\Quorum\ConsensusStrategy;
+    use Symfony\Component\Lock\Store\CombinedStore;
+    use Symfony\Component\Lock\Store\RedisStore;
+
+    $stores = [];
+    foreach (array('server1', 'server2', 'server3') as $server) {
+        $redis= new \Redis();
+        $redis->connect($server);
+
+        $stores[] = new RedisStore($redis);
+    }
+
+    $store = new CombinedStore($stores, new ConsensusStrategy());
+
+Instead of the simple majority strategy (``ConsensusStrategy``) you can use the
```

Comment:
"you can use the ``UnanimousStrategy`` to require the lock" => "an ``UnanimousStrategy`` can be used to require the lock"